<a href="https://colab.research.google.com/github/omar-tueme/ciencia-de-datos-avanzada-mek-enero-2026/blob/main/01_EDA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 3.2 Limpieza y Preparación
### Adriana Ale Huesca - A01022013
### Margarita Ojeda Hernández - A01797688
### Omar Jacobo Tueme Márquez - A00959999


## Preparativos y análisis general de la base de datos

Antes de empezar, vamos a importar las librerias que se van a utilizar a lo largo de este proyecto

In [1020]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns

Posterior a ello, vamos a traernos la base de datos que se encuentra guardada en el repo del proyecto

In [1021]:
url = 'https://raw.githubusercontent.com/omar-tueme/ciencia-de-datos-avanzada-mek-enero-2026/main/datos_info/adolescentes_ensanut2024_w.csv'
df = pd.read_csv(url, sep=';')

Ya se tienen identificadas con anterioridad las variables de interes, entonces hacemos una selección de solo esas variables y creamos un nuevo DataFrame con solo las columnas que nos interesan. El nuevo DataFrame se llama ***df_clabe***

In [1022]:
columns_to_keep = ['FOLIO_I', 'FOLIO_INT', 'd0415', 'edad', 'sexo', 'd0101', 'd0107',
                   'd0108', 'd01a03a', 'd01a03b', 'd01a03c', 'd01a03d', 'd01a08a',
                   'd01a08b', 'd01a08c', 'd01a08d', 'd01a08e', 'd01a08f', 'd01a08g',
                   'd01a08h', 'd01a08i', 'd02o1a', 'd02o1b', 'd02o1c', 'd02o1d',
                   'd0313', 'd0421', 'd0422', 'd0601a', 'd0601b', 'd0601c', 'd0601d',
                   'd0601e', 'd0601f', 'd0601g', 'd0817', 'd0818', 'd06a1', 'd06a2',
                   'd06a3', 'd06a4', 'd06a5', 'd06a6', 'd06a7', 'd06a8', 'd06a9',
                   'd06a10', 'd0701', 'd0702', 'd0709', 'd0309', 'd0801', 'd0810']

df_clean = df[columns_to_keep]

Adicional a ello, ya que el nombre de columnas es cero intuitivo creamos un dicionnario que incluye el nombre de la columna, la pregunta que representa y las posibles respuestas. Este diccionario se volverá critico más adelante cuando hagamos la transformación de nuestras variables para poder hacer un correcto analisis

In [1023]:
dictionary_non_id_variables = {
    'd0415': {
        'Question': 'En comparación con las personas de tu misma edad, ¿tienes dificultad para aprender cosas?',
        1: 'Ninguna dificultad',
        2: 'Cierta dificultad',
        3: 'Mucha dificultad',
        4: 'Le resulta imposible aprender',
    },
    'edad': {
        'Question': 'Edad',
        10: 10,
        11: 11,
        12: 12,
        13: 13,
        14: 14,
        15: 15,
        16: 16,
        17: 17,
        18: 18,
        19: 19,
    },
    'sexo': {
        'Question': 'Sexo',
        1: 'Hombre',
        2: 'Mujer'
    },
    'd0101': {
        'Question': 'Actualmente ¿fumas tabaco...',
        1: 'todos los días?',
        2: 'algunos días?',
        3: 'no fuma actualmente',
        9: 'No responde'
    },
    'd0107': {
        'Question': '¿Actualmente vapeas o consumes cigarros electrónicos...',
        1: 'todos los días?',
        2: 'algunos días?',
        3: 'actualmente no consume',
        4: 'No conoce el cigarro electrónico',
        9: 'No responde'
    },
    'd0108': {
        'Question': 'En los últimos 12 meses, ¿con qué frecuencia tomaste al menos una copa ya sea de vino, cerveza, whisky o cualquier otra bebida que contenga alcohol?',
        1: 'Diario',
        2: 'Semanal',
        3: 'Mensual',
        4: 'Anual',
        5: 'No he consumido en los últimos 12 meses',
        6: 'Nunca he consumido',
        9: 'No responde'
    },
    'd01a03a': {
        'Question': 'Casi todo el mundo toma medicamentos y sustancias por varias razones. Me podrías decir si has tomado, usado, probado opiáceos',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a03b': {
        'Question': 'Casi todo el mundo toma medicamentos y sustancias por varias razones. Me podrías decir si has tomado, usado, probado tranquilizantes',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a03c': {
        'Question': 'Casi todo el mundo toma medicamentos y sustancias por varias razones. Me podrías decir si has tomado, usado, probado sedantes y babitúricos',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a03d': {
        'Question': 'Casi todo el mundo toma medicamentos y sustancias por varias razones. Me podrías decir si has tomado, usado, probado anfetaminas o estimulantes',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08a': {
        'Question': '¿Me podrías decir si has tomado, usado, probado mariguana?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08b': {
        'Question': '¿Me podrías decir si has tomado, usado, probado cocaína?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08c': {
        'Question': '¿Me podrías decir si has tomado, usado, probado crack?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08d': {
        'Question': '¿Me podrías decir si has tomado, usado, probado alucinógenos?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08e': {
        'Question': '¿Me podrías decir si has tomado, usado, probado inhalables?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08f': {
        'Question': '¿Me podrías decir si has tomado, usado, probado heroína?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08g': {
        'Question': '¿Me podrías decir si has tomado, usado, probado estimulantes tipo anfetamínico?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08h': {
        'Question': '¿Me podrías decir si has tomado, usado, probado otras drogas?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd01a08i': {
        'Question': '¿Me podrías decir si has tomado, usado, probado mariguana sintética?',
        1: 'Sí',
        2: 'No',
        9: 'No sabe/No contesta'
    },
    'd02o1a': {
        'Question': '¿Algún médico te ha dicho que tienes diabetes o la azúcar alta en la sangre?',
        1: 'Sí',
        2: 'No',
        9: 'No sé'
    },
    'd02o1b': {
        'Question': '¿Algún médico te ha dicho que tienes la presión alta o hipertensión?',
        1: 'Sí',
        2: 'No',
        9: 'No sé'
    },
    'd02o1c': {
        'Question': '¿Algún médico te ha dicho que tienes el colesterol alto?',
        1: 'Sí',
        2: 'No',
        9: 'No sé'
    },
    'd02o1d': {
        'Question': '¿Algún médico te ha dicho que tienes los triglicéridos altos?',
        1: 'Sí',
        2: 'No',
        9: 'No sé'
    },
    'd0313': {
        'Question': '¿Alguna vez has estado embarazada?',
        1: 'Sí',
        2: 'No',
        9: 'No sé'
    },
    'd0421': {
        'Question': 'Me gustaría saber con qué frecuencia te muestras muy ansioso/a, nervioso/a o preocupado/a.',
        1: 'Diariamente',
        2: 'Semanalmente',
        3: 'Mensualmente',
        4: 'Varias veces al año',
        5: 'Nunca'
    },
    'd0422': {
        'Question': 'Me gustaría saber con qué frecuencia te muestras muy triste o deprimido/a.',
        1: 'Diariamente',
        2: 'Semanalmente',
        3: 'Mensualmente',
        4: 'Varias veces al año',
        5: 'Nunca'
    },
    'd0601a': {
        'Question': 'Durante la última semana... ¿Sentías como si no pudieras quitarte de encima la tristeza?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601b': {
        'Question': 'Durante la última semana... ¿Te costaba concentrarte en lo que estabas haciendo?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601c': {
        'Question': 'Durante la última semana... ¿Te sentiste deprimido/a?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601d': {
        'Question': 'Durante la última semana... ¿Te parecía que todo lo que hacías era un esfuerzo?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601e': {
        'Question': 'Durante la última semana... ¿No dormiste bien?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601f': {
        'Question': 'Durante la última semana... ¿Disfrutaste de la vida?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0601g': {
        'Question': 'Durante la última semana... ¿Te sentiste triste?',
        1: 'Rara vez o nunca (menos de un día)',
        2: 'Pocas veces o alguna vez (1-2 días)',
        3: 'Un número de veces considerables (3-4 días)',
        4: 'Todo el tiempo o la mayoria del tiempo (5-7 días)'
    },
    'd0817': {
        'Question': '¿Alguna vez has pensado en suicidarte?',
        1: 'Sí',
        2: 'No',
        8: 'No responde'
    },
    'd0818': {
        'Question': '¿Con qué frecuencia lo has pensado?',
        1: 'Siempre o muy frecuente',
        2: 'Casi siempre',
        3: 'Algunas veces',
        4: 'Rara vez',
        99: 'No responde'
    },
    'd06a1': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… te ha preocupado engordar?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a2': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… en ocasiones, has comido demasiado?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a3': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has perdido el control sobre lo que comes?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a4': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has vomitado después de comer para bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a5': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has hecho ayunos (dejar de comer por 12 horas o más) para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a6': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has hecho dietas para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a7': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has hecho ejercicio en exceso para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a8': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has usado pastillas para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a9': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has tomado diuréticos (sustancia para perder agua) para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd06a10': {
        'Question': '¿Con qué frecuencia en los últimos 3 meses… has tomado laxantes (sustancia para facilitar la evacuación) para tratar de bajar de peso?',
        1: 'Nunca o casi nunca',
        2: 'Algunas veces',
        3: 'Frecuentemente (2 veces en una semana)',
        4: 'Muy frecuentemente (más de 2 veces en una semana)'
    },
    'd0701': {
        'Question': '¿Sufriste algún daño a tu salud a causa de un accidente en los últimos 12 meses?',
        1: 'Sí',
        2: 'No',
        8: 'No sabe',
        9: 'No responde'
    },
    'd0702': {
        'Question': '¿Cómo fue que te accidentaste?',
        1: 'Choque entre vehículos de 4 o más ruedas',
        2: 'Choque en motocicleta',
        3: 'Choque o atropellamiento como ciclista',
        4: 'Atropellado como peatón',
        5: 'Otros accidentes de transporte: avión, tren, lancha',
        6: 'Caída a nivel de piso',
        7: 'Golpe con objeto, equipo o maquinaria',
        8: 'Caída de un nivel a otro: silla, escalera, etc.',
        9: 'Golpe, rasguño o mordida de persona o animal (caballo, perro, etc.)',
        10: 'Exposición a corriente eléctrica, radiación',
        11: 'Exposición a humo o fuego',
        12: 'Contacto con superficies o sustancias calientes (quemaduras)',
        13: 'Exposición a fuerzas de la naturaleza: rayo, granizo',
        14: 'Picadura de animal ponzoñoso o contacto con plantas venenosas',
        15: 'Envenenamiento accidental: alimentos, medicamentos',
        16: 'Sumersión en depósitos de agua: alberca, cisterna',
        17: 'Otros accidentes que obstruyen la respiración: objeto extraño, alimento',
        18: 'Exposición accidental a otros factores',
        99: 'NS/NR'
    },
    'd0709': {
        'Question': '¿Qué problema de salud permanente te ocasionó el accidente?',
        1: 'Ningún problema',
        2: 'Limitación o dificultad para moverse o caminar (o lo hace con ayuda)',
        3: 'Limitación o dificultad para usar sus brazos y manos',
        4: 'Dificultad para oír',
        5: 'Dificultad para hablar',
        6: 'Dificultad para ver o ceguera',
        7: 'Otra limitación física o mental',
        99: 'NS/NR'
    },
    'd0309': {
        'Question': '¿Fuiste presionada(o) o forzada(o) para tener esa primera relación sexual?',
        1: 'Sí',
        2: 'No',
        9: 'No recuerdo'
    },
    'd0801': {
        'Question': 'En los últimos 12 meses, ¿sufriste algún daño a tu salud por robo, agresión o violencia?',
        1: 'Sí',
        2: 'No',
        9: 'No responde'
    },
    'd0810': {
        'Question': '¿A lo largo de tu vida, alguien te manoseó, tocó o acarició alguna parte de tu cuerpo o tuvo relaciones sexuales contigo cuando eras muy pequeño/a?',
        1: 'Sí, antes de los 12',
        2: 'Sí, cuando tenía 12 o más años',
        3: 'No, nunca',
        8: 'No recuerda',
        9: 'No responde'
    }
}

Empezamos con información general de DataFrame, se van a hacer varias transformaciones pero esto nos sirve para ver como está construido el DataFrame y que tenemos un total de 53 columnas y 3,730 filas. Cuando nos metemos a estudiar cada una de las variables, encontramos algunas concideraciones importantes (otra vez, esto es a gran escala, más adelante nos enfocaremos en entender cada una de nuestras variables)

* Los "nulls" se encuentrán guardados como " " por lo que los transformaremos más adelante en nulls. Éstas columnas con " " son de tipo objecto

* La variable objetivo ***d0415*** (En comparación con las personas de tu misma edad, ¿tienes dificultad para aprender cosas?) tiene 690 nulls, estas observaciones se tendrán que quitar del análisis ya que no aportan nada

* Hay preguntas como ***d0313*** (¿Alguna vez has estado embarazada?), ***d0818*** (¿Con qué frecuencia lo has pensado [suicidarte]?), ***d0702*** (¿Cómo fue que te accidentaste?), ***d0709*** (¿Qué problema de salud permanente te ocasionó el accidente?) y ***d0309*** (¿Fuiste presionada(o) o forzada(o) para tener esa primera relación sexual?) con una cantidad muy elevada de valores faltantes (por la misma naturaleza de la pregunta). Se evaluara con precación si dichas columan deberían de ser excluidas del análisis


In [1024]:
df_clean.head()

,FOLIO_I,FOLIO_INT,d0415,edad,sexo,d0101,d0107,d0108,d01a03a,d01a03b,...,d06a7,d06a8,d06a9,d06a10,d0701,d0702,d0709,d0309,d0801,d0810
0,2024_01001001,2024_01001001_05,1,11,2,3,4,6,,,...,1,1,1,1,2,,,,2,3
1,2024_01001006,2024_01001006_03,,19,2,3,3,3,2,2,...,1,1,1,1,2,,,2,2,1
2,2024_01001009,2024_01001009_03,1,15,1,3,3,2,2,2,...,1,1,1,1,2,,,,2,9
3,2024_01001024,2024_01001024_03,,19,1,2,3,2,2,2,...,1,1,1,1,2,,,,2,3
4,2024_01001026,2024_01001026_03,,19,1,3,2,3,2,2,...,1,1,1,1,2,,,2,2,3


In [1025]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3730 entries, 0 to 3729
Data columns (total 53 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   FOLIO_I    3730 non-null   object
 1   FOLIO_INT  3730 non-null   object
 2   d0415      3730 non-null   object
 3   edad       3730 non-null   int64 
 4   sexo       3730 non-null   int64 
 5   d0101      3730 non-null   int64 
 6   d0107      3730 non-null   int64 
 7   d0108      3730 non-null   int64 
 8   d01a03a    3730 non-null   object
 9   d01a03b    3730 non-null   object
 10  d01a03c    3730 non-null   object
 11  d01a03d    3730 non-null   object
 12  d01a08a    3730 non-null   object
 13  d01a08b    3730 non-null   object
 14  d01a08c    3730 non-null   object
 15  d01a08d    3730 non-null   object
 16  d01a08e    3730 non-null   object
 17  d01a08f    3730 non-null   object
 18  d01a08g    3730 non-null   object
 19  d01a08h    3730 non-null   object
 20  d01a08i    3730 non-null   obj

In [1026]:
for col in df_clean.columns:
    print(f'Variable: {col}.\nNumber unique values: {df_clean[col].nunique()}.\n{df_clean[col].value_counts()}\n\n')

Variable: FOLIO_I.
Number unique values: 3730.
FOLIO_I
2024_32056030    1
2024_01001001    1
2024_32052016    1
2024_32052015    1
2024_32052014    1
                ..
2024_01001032    1
2024_01001028    1
2024_01001026    1
2024_01001024    1
2024_01001009    1
Name: count, Length: 3730, dtype: int64


Variable: FOLIO_INT.
Number unique values: 3730.
FOLIO_INT
2024_32056030_03    1
2024_01001001_05    1
2024_32052016_04    1
2024_32052015_05    1
2024_32052014_04    1
                   ..
2024_01001032_02    1
2024_01001028_04    1
2024_01001026_03    1
2024_01001024_03    1
2024_01001009_03    1
Name: count, Length: 3730, dtype: int64


Variable: d0415.
Number unique values: 5.
d0415
1    2461
      690
2     532
3      46
4       1
Name: count, dtype: int64


Variable: edad.
Number unique values: 10.
edad
11    423
13    421
15    398
10    387
12    378
17    366
18    345
19    344
14    335
16    333
Name: count, dtype: int64


Variable: sexo.
Number unique values: 2.
sexo
1   

## Limpieza de la base de datos

Empezaremos categorizar todos los " " como nulls

Después de hacerlo nos damos cuenta que si tenemos nulls, no como se vió en el primer df_clean.info

Vamos a aprovechar para remover las columnas de folio ya que no sirven para el análisis

Por último se va a convertir todas las columnas a Int64 ya que en realidad lo que se tienen son números y no texto

In [1027]:
df_clean = df_clean.replace(' ', np.nan)

In [1028]:
df_clean = df_clean.drop(columns=['FOLIO_I', 'FOLIO_INT'])

df_clean = df_clean.astype('Int64')

In [1029]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3730 entries, 0 to 3729
Data columns (total 51 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   d0415    3040 non-null   Int64
 1   edad     3730 non-null   Int64
 2   sexo     3730 non-null   Int64
 3   d0101    3730 non-null   Int64
 4   d0107    3730 non-null   Int64
 5   d0108    3730 non-null   Int64
 6   d01a03a  2927 non-null   Int64
 7   d01a03b  2927 non-null   Int64
 8   d01a03c  2927 non-null   Int64
 9   d01a03d  2927 non-null   Int64
 10  d01a08a  2927 non-null   Int64
 11  d01a08b  2927 non-null   Int64
 12  d01a08c  2927 non-null   Int64
 13  d01a08d  2927 non-null   Int64
 14  d01a08e  2927 non-null   Int64
 15  d01a08f  2927 non-null   Int64
 16  d01a08g  2927 non-null   Int64
 17  d01a08h  2927 non-null   Int64
 18  d01a08i  2927 non-null   Int64
 19  d02o1a   3730 non-null   Int64
 20  d02o1b   3730 non-null   Int64
 21  d02o1c   3730 non-null   Int64
 22  d02o1d   3730 non-null  

Posterior, se van a remover todas las observaciones que no tienen un valor en nuestra variable objetivo. Esto porque dichas observaciones no estarían aportando a este análisis en especifico. Esto la vamos a guardar en un nuevo DataFrame llamado ***df_clean1***

Ésto nos deja con un total de 53 columnas y 3040 renglonse

In [1030]:
df_clean1['d0810'].unique()

<IntegerArray>
[3, 9, 1, 8, 2]
Length: 5, dtype: Int64

In [1031]:
df_clean1 = df_clean.dropna(subset=['d0415']).reset_index(drop=True)

df_clean1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3040 entries, 0 to 3039
Data columns (total 51 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   d0415    3040 non-null   Int64
 1   edad     3040 non-null   Int64
 2   sexo     3040 non-null   Int64
 3   d0101    3040 non-null   Int64
 4   d0107    3040 non-null   Int64
 5   d0108    3040 non-null   Int64
 6   d01a03a  2237 non-null   Int64
 7   d01a03b  2237 non-null   Int64
 8   d01a03c  2237 non-null   Int64
 9   d01a03d  2237 non-null   Int64
 10  d01a08a  2237 non-null   Int64
 11  d01a08b  2237 non-null   Int64
 12  d01a08c  2237 non-null   Int64
 13  d01a08d  2237 non-null   Int64
 14  d01a08e  2237 non-null   Int64
 15  d01a08f  2237 non-null   Int64
 16  d01a08g  2237 non-null   Int64
 17  d01a08h  2237 non-null   Int64
 18  d01a08i  2237 non-null   Int64
 19  d02o1a   3040 non-null   Int64
 20  d02o1b   3040 non-null   Int64
 21  d02o1c   3040 non-null   Int64
 22  d02o1d   3040 non-null  

Por último, para que el proceso de limpieza sea mucho más sencillo se va a sustituir los valores de las celdas y los títulos de las columnas con lo que se tiene en el diccionario. Esto es meramente transitorio para poder hacer una limpieza de datos más sencilla

Las siguientes lineas de código fueron escritas con apoyo por parte de ChatGPT

In [1032]:
"""
NOT USED -- TO CHANGE COLUMN NAMES BUT IT MAKES SUBSEQUENT CLEANING VERY DIFFICULT

rename_dict = {
    var: info["Question"]
    for var, info in dictionary_non_id_variables.items()
    if var in df_clean1.columns
}

df_clean1 = df_clean1.rename(columns=rename_dict)
"""

'\nNOT USED -- TO CHANGE COLUMN NAMES BUT IT MAKES SUBSEQUENT CLEANING VERY DIFFICULT\n\nrename_dict = {\n    var: info["Question"]\n    for var, info in dictionary_non_id_variables.items()\n    if var in df_clean1.columns\n}\n\ndf_clean1 = df_clean1.rename(columns=rename_dict)\n'

In [1033]:
for var, info in dictionary_non_id_variables.items():
    if var in df_clean1.columns:
        mapping = {k: v for k, v in info.items() if isinstance(k, int)}

        df_clean1[var] = df_clean1[var].map(mapping)

In [1034]:
df_clean1

,d0415,edad,sexo,d0101,d0107,d0108,d01a03a,d01a03b,d01a03c,d01a03d,...,d06a7,d06a8,d06a9,d06a10,d0701,d0702,d0709,d0309,d0801,d0810
0,Ninguna dificultad,11,Mujer,no fuma actualmente,No conoce el cigarro electrónico,Nunca he consumido,NaN,NaN,NaN,NaN,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
1,Ninguna dificultad,15,Hombre,no fuma actualmente,actualmente no consume,Semanal,No,No,No,No,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,No responde
2,Ninguna dificultad,13,Hombre,no fuma actualmente,actualmente no consume,Nunca he consumido,No,No,No,No,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
3,Ninguna dificultad,10,Mujer,no fuma actualmente,No conoce el cigarro electrónico,Nunca he consumido,NaN,NaN,NaN,NaN,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
4,Ninguna dificultad,13,Mujer,no fuma actualmente,actualmente no consume,Nunca he consumido,No,No,No,No,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,Ninguna dificultad,13,Hombre,no fuma actualmente,actualmente no consume,Nunca he consumido,No,No,No,No,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
3036,Ninguna dificultad,13,Mujer,no fuma actualmente,actualmente no consume,Nunca he consumido,No,No,No,No,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
3037,Ninguna dificultad,11,Mujer,no fuma actualmente,actualmente no consume,No he consumido en los últimos 12 meses,NaN,NaN,NaN,NaN,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"
3038,Ninguna dificultad,11,Mujer,no fuma actualmente,actualmente no consume,Nunca he consumido,NaN,NaN,NaN,NaN,...,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,Nunca o casi nunca,No,NaN,NaN,NaN,No,"No, nunca"


Con la base de datos más entendible, vamos a proceder a enfocarnos en cada una de las subsecciones y en realizar la limpieza y preparación pertinente. La intensión es convertir la mayoría de las variables en variables dummies para poder hacer un análisis categórico

### Variable dependiente

Empezemos por recordar cómo está construida la variable dependiente. 2,461 de 3,040 (80%) dice no tener ninguna dificultad aprendiendo en comparación a personas de su misma edad

Dado el bajo volumen en las respuestas de "mucha dificultad" y "le resulta imposible aprender", se van a agrupar con la respuesta" cierta dificultad" para construir la variable dicotómica que compara ninguna dificultad de aprender versus alguna dificultad para aprender

Todas estas transformación se irán poniendo en un nuevo DataFrame llamado ***df_final*** que es el que se utilizará para realizar el análisis

In [1035]:
df_clean1['d0415'].value_counts(dropna=False)

,count
d0415,
Ninguna dificultad,2461
Cierta dificultad,532
Mucha dificultad,46
Le resulta imposible aprender,1


In [1036]:
df_final = pd.DataFrame()

df_final['Dificultad para aprender'] = (df_clean1['d0415'] != 'Ninguna dificultad').astype(int)

df_final['Dificultad para aprender'].value_counts()

,count
Dificultad para aprender,
0,2461
1,579


In [1037]:
df_final

,Dificultad para aprender
0,0
1,0
2,0
3,0
4,0
...,...
3035,0
3036,0
3037,0
3038,0


### Variables de identidad

Posterior a la variable dependiente, tenemos las variables de identidad (sexo y edad), recordemos su distribución. De ahora en adelante siempre se correra crosstab contra la variable dependiendo para ir observando patrones

Parece ser que al limpiar por nuestra variable dependiente se impacto desproporcionalmente a los jovenes de edad entre los 18 y 19 años. Al revisar la encuesta, existe una leyenda que indica al entrevistador pasarse la pregunta de interes para las personas de esa edad. Es por ello que, por consistencia en los resultados, se eliminarán los 6 registros de las personas con edad entre 18 y 19 años -- dicha limpieza se hará hasta el final de armar df_final para evitar que los datos no empaten al ir construyendo el DataFrame de manera correcta con las siguientes subsecciones

Respecto al sexo, se mantiene una distribución muy cercana a 50/50, por lo que se mantiene la variable como está. Ambas variables se guardaran en ***df_final***, edad siendo una variable discreta y sexo siendo una variable dicotómica

In [1038]:
df_clean1['edad'].value_counts(dropna=False).sort_index()

,count
edad,
10,387
11,423
12,378
13,421
14,335
15,397
16,333
17,360
18,5


In [1039]:
df_clean['edad'].value_counts(dropna=False).sort_index()

,count
edad,
10,387
11,423
12,378
13,421
14,335
15,398
16,333
17,366
18,345


In [1040]:
pd.crosstab(
    df_clean1['edad'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
edad,,,,
10,88,0,7,292
11,89,1,4,329
12,55,0,12,311
13,94,0,6,321
14,56,0,5,274
15,52,0,7,338
16,46,0,3,284
17,52,0,2,306
18,0,0,0,5


In [1041]:
df_clean1['sexo'].value_counts(dropna=False).sort_index()

,count
sexo,
Hombre,1541
Mujer,1499


In [1042]:
pd.crosstab(
    df_clean1['sexo'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
sexo,,,,
Hombre,261,0,22,1258
Mujer,271,1,24,1203


In [1043]:
df_final['Edad'] = df_clean1['edad']

df_final['Edad'].value_counts(dropna=False).sort_index()

,count
Edad,
10,387
11,423
12,378
13,421
14,335
15,397
16,333
17,360
18,5


In [1044]:
df_final['Es mujer'] = (df_clean1['sexo'] == 'Mujer').astype(int)

df_final['Es mujer'].value_counts(dropna=False)

,count
Es mujer,
0,1541
1,1499


In [1045]:
df_final

,Dificultad para aprender,Edad,Es mujer
0,0,11,1
1,0,15,0
2,0,13,0
3,0,10,1
4,0,13,1
...,...,...,...
3035,0,13,0
3036,0,13,1
3037,0,11,1
3038,0,11,1


### Variables de drogas - Tabaco

Recordemos como se distribuyen estas variables. Dado lo que se vió arriba con la variable edad en donde parece que hay preguntas que no se hace a cierta edad, vamos a correr crosstab contra la variable objetivo y contra edad para ser precavidos en todos los futuros chequeos

Parecen haber pocos casos de jovenes que fuman (buenas noticias!), para generar la variable dummy se harán 3 agrupaciones

* "No responde" y "No conoce el cigarro electrónico" se le asignará la etiqueta de no fuma actualmente dado que esta respuesta es la moda

* "algunos días?" y "todos los días?" se agruparan para representar que el joven fuma (sin importar la frecuencia)

* Para finalizar, ambas columnas se colapsaran a 1 sola, si una persona fuma o consume cigarro electronico se considerará fumador. Si no consume ninguno de los dos, se considerará un no fumador

In [1046]:
df_clean1['d0101'].value_counts(dropna=False).sort_index()

,count
d0101,
No responde,38
algunos días?,43
no fuma actualmente,2949
todos los días?,10


In [1047]:
pd.crosstab(
    df_clean1['d0101'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0101,,,,
No responde,7,0,0,31
algunos días?,5,0,0,38
no fuma actualmente,517,1,46,2385
todos los días?,3,0,0,7


In [1048]:
pd.crosstab(
    df_clean1['d0101'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0101,,,,,,,,,,
No responde,3,7,3,9,3,2,7,4,0,0
algunos días?,1,0,3,1,4,7,10,17,0,0
no fuma actualmente,383,416,372,410,327,386,315,334,5,1
todos los días?,0,0,0,1,1,2,1,5,0,0


In [1049]:
df_clean1['d0107'].value_counts(dropna=False).sort_index()

,count
d0107,
No conoce el cigarro electrónico,1084
No responde,15
actualmente no consume,1875
algunos días?,64
todos los días?,2


In [1050]:
pd.crosstab(
    df_clean1['d0107'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0107,,,,
No conoce el cigarro electrónico,198,1,24,861
No responde,3,0,0,12
actualmente no consume,317,0,22,1536
algunos días?,14,0,0,50
todos los días?,0,0,0,2


In [1051]:
pd.crosstab(
    df_clean1['d0107'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0107,,,,,,,,,,
No conoce el cigarro electrónico,229,223,147,127,101,102,75,76,3,1
No responde,0,4,0,1,1,1,5,3,0,0
actualmente no consume,157,195,227,282,224,286,240,262,2,0
algunos días?,1,1,4,11,9,8,13,17,0,0
todos los días?,0,0,0,0,0,0,0,2,0,0


In [1052]:
pd.crosstab(
    df_clean1['d0107'],
    df_clean1['d0101'],
    dropna=False
)

d0101,No responde,algunos días?,no fuma actualmente,todos los días?
d0107,,,,
No conoce el cigarro electrónico,11,8,1063,2
No responde,11,0,4,0
actualmente no consume,15,24,1830,6
algunos días?,1,10,51,2
todos los días?,0,1,1,0


In [1053]:
df_final['Fuma'] = ((df_clean1['d0101'] == 'algunos días?') |
 (df_clean1['d0101'] == 'todos los días?') |
  (df_clean1['d0107'] == 'algunos días?') |
   (df_clean1['d0101'] == 'todos los días?')).astype(int)

df_final['Fuma'].value_counts(dropna=False)

,count
Fuma,
0,2935
1,105


In [1054]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma
0,0,11,1,0
1,0,15,0,0
2,0,13,0,0
3,0,10,1,0
4,0,13,1,0
...,...,...,...,...
3035,0,13,0,0
3036,0,13,1,0
3037,0,11,1,0
3038,0,11,1,0


### Variables de drogas - Alcohol

Recordemos como se distribuye esta variable. A pesar de tener datos de frecuencia de consumo, convertiremos la variable en una variable dicotómica en donde haremos el corte a nivel semanal. De este modo los entrevistados que consuman alcohol de manera dairia o semanal se les asignara un 1 mientras que los que consuman con una mayor frecuencia o no consuman se les asignará un 0

In [1055]:
df_clean1['d0108'].value_counts(dropna=False).sort_index()

,count
d0108,
Anual,231
Diario,3
Mensual,96
No he consumido en los últimos 12 meses,660
No responde,7
Nunca he consumido,2001
Semanal,42


In [1056]:
pd.crosstab(
    df_clean1['d0108'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0108,,,,
Anual,44,0,0,187
Diario,0,0,0,3
Mensual,11,0,1,84
No he consumido en los últimos 12 meses,107,0,11,542
No responde,0,0,1,6
Nunca he consumido,364,1,32,1604
Semanal,6,0,1,35


In [1057]:
pd.crosstab(
    df_clean1['d0108'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0108,,,,,,,,,,
Anual,2,8,6,23,27,52,47,65,1,0
Diario,0,0,0,0,0,1,1,1,0,0
Mensual,0,1,3,2,7,19,31,33,0,0
No he consumido en los últimos 12 meses,84,89,90,106,71,81,68,70,1,0
No responde,0,2,0,1,1,0,2,1,0,0
Nunca he consumido,300,323,278,286,227,238,173,172,3,1
Semanal,1,0,1,3,2,6,11,18,0,0


In [1058]:
"""
NOT USED -- WE DECIDED AGAINS THIS APPROACH, KEEPING IT FOR SAFEKEEPING

def scale_alcohol(row):
    x = row['d0108']
    if x == 'Diario':
        return 1
    elif x == 'Semanal':
        return 52/365
    elif x == 'Mensual':
        return 12/365
    elif x == 'Anual':
        return 1/365
    else:
        return 0

df_final['Alcohol'] = df_clean1.apply(scale_alcohol, axis=1)

df_final['Alcohol'].value_counts()
"""

"\nNOT USED -- WE DECIDED AGAINS THIS APPROACH, KEEPING IT FOR SAFEKEEPING\n\ndef scale_alcohol(row):\n    x = row['d0108']\n    if x == 'Diario':\n        return 1\n    elif x == 'Semanal':\n        return 52/365\n    elif x == 'Mensual':\n        return 12/365\n    elif x == 'Anual':\n        return 1/365\n    else:\n        return 0\n\ndf_final['Alcohol'] = df_clean1.apply(scale_alcohol, axis=1)\n\ndf_final['Alcohol'].value_counts()\n"

In [1059]:
df_final['Alcohol'] = ((df_clean1['d0108'] == 'Diario') | (df_clean1['d0108'] == 'Semanal')).astype(int)

df_final['Alcohol'].value_counts(dropna=False)

,count
Alcohol,
0,2995
1,45


In [1060]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol
0,0,11,1,0,0
1,0,15,0,0,1
2,0,13,0,0,0
3,0,10,1,0,0
4,0,13,1,0,0
...,...,...,...,...,...
3035,0,13,0,0,0
3036,0,13,1,0,0
3037,0,11,1,0,0
3038,0,11,1,0,0


### Variables de drogas - Drogas Controladas

Recordemos como se distribuyen estas variables. Encontramos algo particular en esta categoria, tenemos 803 nulls pero la mayoria se concentran en los años de 10 y 11 años. El motivo de ello es que durante la encuesta se da la instrucción de brincarse esta sección para niños entre 10 y 11 años de edad. Dado que quitar estas variables significaría quitar a 795 de los 810 encuestados entre 10 y 11 años (98%) se decide mantener las variables pero especificar 2 modelos distintos. Para niños entre 10 y 11 años no se incluirá esta variable mientras que para niños de 12 años a 17 años si se considerará. Sin embargo, aún hay nulls en otras edades -- 7 para adolescentes de 12 años (1.9%) y 1 para adolescentes de 15 años (0.3%) -- lo más seguro es que error en la captura de datos. Dado lo anterior y lo que se observa en los datos y las preguntas, se decide hacer las siguientes modificaciones

* Dado que asumimos que las encuestas con null en edades mayores a 11 años son errores de caputura y dado que representan un pequeño % de la población, se decide borrar estas filas (igual que con las edades de 18 y 19 años, esto se hará al final de construir df_final)

* No se considerarán estas variables para las edades de 10 y 11 años y se especificarán 2 modelos distintos (10 - 11 años y 12 - 17 años)

* "No sabe/No contesta" se transformará a "No"

* Se combinarán similar a como se hizo con el tabaco tanto d01a03b como d01a03c ya que ambos tratan de tranquilizantes / sedantes

In [1061]:
df_clean1['d01a03a'].value_counts(dropna=False).sort_index()

,count
d01a03a,
No,2191
No sabe/No contesta,24
Sí,22
NaN,803


In [1062]:
pd.crosstab(
    df_clean1['d01a03a'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a03a,,,,
No,347,0,33,1811
No sabe/No contesta,5,0,0,19
Sí,6,0,2,14
NaN,174,1,11,617


In [1063]:
pd.crosstab(
    df_clean1['d01a03a'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a03a,,,,,,,,,,
No,3,11,363,413,333,385,324,353,5,1
No sabe/No contesta,0,0,3,4,0,6,7,4,0,0
Sí,0,1,5,4,2,5,2,3,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1064]:
df_clean1['d01a03b'].value_counts(dropna=False).sort_index()

,count
d01a03b,
No,2190
No sabe/No contesta,16
Sí,31
NaN,803


In [1065]:
pd.crosstab(
    df_clean1['d01a03b'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a03b,,,,
No,350,0,33,1807
No sabe/No contesta,4,0,0,12
Sí,4,0,2,25
NaN,174,1,11,617


In [1066]:
pd.crosstab(
    df_clean1['d01a03b'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a03b,,,,,,,,,,
No,3,11,366,415,330,387,325,347,5,1
No sabe/No contesta,0,0,2,1,1,3,3,6,0,0
Sí,0,1,3,5,4,6,5,7,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1067]:
df_clean1['d01a03c'].value_counts(dropna=False).sort_index()

,count
d01a03c,
No,2206
No sabe/No contesta,15
Sí,16
NaN,803


In [1068]:
pd.crosstab(
    df_clean1['d01a03c'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a03c,,,,
No,350,0,34,1822
No sabe/No contesta,3,0,0,12
Sí,5,0,1,10
NaN,174,1,11,617


In [1069]:
pd.crosstab(
    df_clean1['d01a03c'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a03c,,,,,,,,,,
No,3,10,368,418,329,391,328,353,5,1
No sabe/No contesta,0,1,3,1,3,2,1,4,0,0
Sí,0,1,0,2,3,3,4,3,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1070]:
pd.crosstab(
    df_clean1['d01a03c'],
    df_clean1['d01a03b'],
    dropna=False
)

d01a03b,No,No sabe/No contesta,Sí,NaN
d01a03c,,,,
No,2178,5,23,0
No sabe/No contesta,3,10,2,0
Sí,9,1,6,0
NaN,0,0,0,803


In [1071]:
df_clean1['d01a03d'].value_counts(dropna=False).sort_index()

,count
d01a03d,
No,2217
No sabe/No contesta,12
Sí,8
NaN,803


In [1072]:
pd.crosstab(
    df_clean1['d01a03d'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a03d,,,,
No,354,0,33,1830
No sabe/No contesta,2,0,0,10
Sí,2,0,2,4
NaN,174,1,11,617


In [1073]:
pd.crosstab(
    df_clean1['d01a03d'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a03d,,,,,,,,,,
No,3,10,366,418,332,394,332,356,5,1
No sabe/No contesta,0,1,4,0,1,2,1,3,0,0
Sí,0,1,1,3,2,0,0,1,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1074]:
df_final['Opiáceos'] = np.where(
    df_clean1['d01a03a'].isna(),
    pd.NA,
    (df_clean1['d01a03a'] == 'Sí').astype(int)
)

df_final['Opiáceos'].value_counts(dropna=False)

,count
Opiáceos,
0,2215
<NA>,803
1,22


In [1075]:
df_final['Tranquilizantes'] = np.where(
    df_clean1['d01a03b'].isna() & df_clean1['d01a03c'].isna(),
    pd.NA,
    ((df_clean1['d01a03b'] == 'Sí') | (df_clean1['d01a03c'] == 'Sí')).astype(int)
)

df_final['Tranquilizantes'].value_counts(dropna=False)

,count
Tranquilizantes,
0,2196
<NA>,803
1,41


In [1076]:
df_final['Anfetaminas'] = np.where(
    df_clean1['d01a03d'].isna(),
    pd.NA,
    (df_clean1['d01a03d'] == 'Sí').astype(int)
)

df_final['Anfetaminas'].value_counts(dropna=False)

,count
Anfetaminas,
0,2229
<NA>,803
1,8


In [1077]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas
0,0,11,1,0,0,<NA>,<NA>,<NA>
1,0,15,0,0,1,0,0,0
2,0,13,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>
4,0,13,1,0,0,0,0,0
...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>
3038,0,11,1,0,0,<NA>,<NA>,<NA>


### Variables de drogas - Drogas Ilicitas

Recordemos como se distribuyen estas variables. Aquí podemos observar el mismo patrón que la sección anterior en donde estas preguntas no se le hicieron a niños entre 10 y 11 años. Se seguirá una estrategia similar de utilizar estas variables en el modelo para niños mayores de 12 años y no utilizarlas para niños entre 10 y 11 años. En este caso se colapsarán todas estas preguntas en una sola variable que considerará el uso de cualquier sustancia ilícita. Similar a como se ha hecho con anterioridad, "No sabe/No contesta" se tratará como un "No"

In [1078]:
df_clean1['d01a08a'].value_counts(dropna=False).sort_index()

,count
d01a08a,
No,2173
No sabe/No contesta,4
Sí,60
NaN,803


In [1079]:
pd.crosstab(
    df_clean1['d01a08a'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08a,,,,
No,346,0,33,1794
No sabe/No contesta,2,0,0,2
Sí,10,0,2,48
NaN,174,1,11,617


In [1080]:
pd.crosstab(
    df_clean1['d01a08a'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08a,,,,,,,,,,
No,3,12,366,418,332,384,314,338,5,1
No sabe/No contesta,0,0,1,1,0,1,0,1,0,0
Sí,0,0,4,2,3,11,19,21,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1081]:
df_clean1['d01a08b'].value_counts(dropna=False).sort_index()

,count
d01a08b,
No,2218
No sabe/No contesta,9
Sí,10
NaN,803


In [1082]:
pd.crosstab(
    df_clean1['d01a08b'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08b,,,,
No,354,0,35,1829
No sabe/No contesta,2,0,0,7
Sí,2,0,0,8
NaN,174,1,11,617


In [1083]:
pd.crosstab(
    df_clean1['d01a08b'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08b,,,,,,,,,,
No,3,12,370,416,335,394,329,353,5,1
No sabe/No contesta,0,0,1,4,0,1,1,2,0,0
Sí,0,0,0,1,0,1,3,5,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1084]:
df_clean1['d01a08c'].value_counts(dropna=False).sort_index()

,count
d01a08c,
No,2224
No sabe/No contesta,11
Sí,2
NaN,803


In [1085]:
pd.crosstab(
    df_clean1['d01a08c'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08c,,,,
No,356,0,34,1834
No sabe/No contesta,1,0,1,9
Sí,1,0,0,1
NaN,174,1,11,617


In [1086]:
pd.crosstab(
    df_clean1['d01a08c'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08c,,,,,,,,,,
No,3,12,368,418,334,394,331,358,5,1
No sabe/No contesta,0,0,3,3,1,1,2,1,0,0
Sí,0,0,0,0,0,1,0,1,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1087]:
df_clean1['d01a08d'].value_counts(dropna=False).sort_index()

,count
d01a08d,
No,2224
No sabe/No contesta,10
Sí,3
NaN,803


In [1088]:
pd.crosstab(
    df_clean1['d01a08d'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08d,,,,
No,357,0,35,1832
No sabe/No contesta,1,0,0,9
Sí,0,0,0,3
NaN,174,1,11,617


In [1089]:
pd.crosstab(
    df_clean1['d01a08d'],
    df_clean1['edad'],
    dropna=False
)

/usr/local/lib/python3.12/dist-packages/IPython/core/displayhook.py:275: UserWarning: Output cache limit (currently 1000 entries) hit.
Flushing oldest 200 entries.
  warn('Output cache limit (currently {sz} entries) hit.\n'


edad,10,11,12,13,14,15,16,17,18,19
d01a08d,,,,,,,,,,
No,3,12,367,419,335,395,330,357,5,1
No sabe/No contesta,0,0,4,2,0,1,1,2,0,0
Sí,0,0,0,0,0,0,2,1,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1090]:
df_clean1['d01a08e'].value_counts(dropna=False).sort_index()

,count
d01a08e,
No,2224
No sabe/No contesta,8
Sí,5
NaN,803


In [1091]:
pd.crosstab(
    df_clean1['d01a08e'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08e,,,,
No,355,0,34,1835
No sabe/No contesta,1,0,0,7
Sí,2,0,1,2
NaN,174,1,11,617


In [1092]:
pd.crosstab(
    df_clean1['d01a08e'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08e,,,,,,,,,,
No,3,12,368,421,332,394,331,357,5,1
No sabe/No contesta,0,0,3,0,2,0,1,2,0,0
Sí,0,0,0,0,1,2,1,1,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1093]:
df_clean1['d01a08f'].value_counts(dropna=False).sort_index()

,count
d01a08f,
No,2229
No sabe/No contesta,8
NaN,803


In [1094]:
pd.crosstab(
    df_clean1['d01a08f'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08f,,,,
No,357,0,35,1837
No sabe/No contesta,1,0,0,7
NaN,174,1,11,617


In [1095]:
pd.crosstab(
    df_clean1['d01a08f'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08f,,,,,,,,,,
No,3,12,367,421,335,396,331,358,5,1
No sabe/No contesta,0,0,4,0,0,0,2,2,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1096]:
df_clean1['d01a08g'].value_counts(dropna=False).sort_index()

,count
d01a08g,
No,2225
No sabe/No contesta,7
Sí,5
NaN,803


In [1097]:
pd.crosstab(
    df_clean1['d01a08g'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08g,,,,
No,356,0,34,1835
No sabe/No contesta,1,0,0,6
Sí,1,0,1,3
NaN,174,1,11,617


In [1098]:
pd.crosstab(
    df_clean1['d01a08g'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08g,,,,,,,,,,
No,3,12,369,420,334,394,332,355,5,1
No sabe/No contesta,0,0,2,0,1,1,1,2,0,0
Sí,0,0,0,1,0,1,0,3,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1099]:
df_clean1['d01a08h'].value_counts(dropna=False).sort_index()

,count
d01a08h,
No,2231
No sabe/No contesta,4
Sí,2
NaN,803


In [1100]:
pd.crosstab(
    df_clean1['d01a08h'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08h,,,,
No,357,0,35,1839
No sabe/No contesta,1,0,0,3
Sí,0,0,0,2
NaN,174,1,11,617


In [1101]:
pd.crosstab(
    df_clean1['d01a08h'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08h,,,,,,,,,,
No,3,12,369,421,335,396,330,359,5,1
No sabe/No contesta,0,0,2,0,0,0,1,1,0,0
Sí,0,0,0,0,0,0,2,0,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1102]:
df_clean1['d01a08i'].value_counts(dropna=False).sort_index()

,count
d01a08i,
No,2225
No sabe/No contesta,8
Sí,4
NaN,803


In [1103]:
pd.crosstab(
    df_clean1['d01a08i'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d01a08i,,,,
No,357,0,35,1833
No sabe/No contesta,1,0,0,7
Sí,0,0,0,4
NaN,174,1,11,617


In [1104]:
pd.crosstab(
    df_clean1['d01a08i'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d01a08i,,,,,,,,,,
No,3,12,369,419,334,395,330,357,5,1
No sabe/No contesta,0,0,2,2,0,1,1,2,0,0
Sí,0,0,0,0,1,0,2,1,0,0
NaN,384,411,7,0,0,1,0,0,0,0


In [1105]:
df_final['Drogas ilícitas'] = np.where(
    df_clean1['d01a08a'].isna() & df_clean1['d01a08b'].isna() & df_clean1['d01a08c'].isna() &
    df_clean1['d01a08d'].isna() & df_clean1['d01a08e'].isna() & df_clean1['d01a08f'].isna() &
    df_clean1['d01a08g'].isna() & df_clean1['d01a08h'].isna() & df_clean1['d01a08i'].isna(),
    pd.NA,
    ((df_clean1['d01a08a'] == 'Sí') | (df_clean1['d01a08b'] == 'Sí') | (df_clean1['d01a08c'] == 'Sí') |
     (df_clean1['d01a08d'] == 'Sí') | (df_clean1['d01a08e'] == 'Sí') | (df_clean1['d01a08f'] == 'Sí') |
     (df_clean1['d01a08g'] == 'Sí') | (df_clean1['d01a08h'] == 'Sí') | (df_clean1['d01a08i'] == 'Sí')).astype(int)
)

df_final['Drogas ilícitas'].value_counts(dropna=False)

,count
Drogas ilícitas,
0,2173
<NA>,803
1,64


In [1106]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>
1,0,15,0,0,1,0,0,0,0
2,0,13,0,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>
4,0,13,1,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>


### Variables de Enfermedades Crónicas

Recordemos como se distribuyen estas variables. Dado que cada enfermedad crónica es muy distinta la una de la otra se mantendrá una variable por cada una. Los "No sé" se tratarán como "No"

In [1107]:
df_clean1['d02o1a'].value_counts(dropna=False).sort_index()

,count
d02o1a,
No,3009
No sé,15
Sí,16


In [1108]:
pd.crosstab(
    df_clean1['d02o1a'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d02o1a,,,,
No,526,1,46,2436
No sé,3,0,0,12
Sí,3,0,0,13


In [1109]:
pd.crosstab(
    df_clean1['d02o1a'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d02o1a,,,,,,,,,,
No,384,420,373,418,332,391,327,358,5,1
No sé,2,2,3,3,1,2,2,0,0,0
Sí,1,1,2,0,2,4,4,2,0,0


In [1110]:
df_clean1['d02o1b'].value_counts(dropna=False).sort_index()

,count
d02o1b,
No,2998
No sé,12
Sí,30


In [1111]:
pd.crosstab(
    df_clean1['d02o1b'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d02o1b,,,,
No,527,1,45,2425
No sé,1,0,1,10
Sí,4,0,0,26


In [1112]:
pd.crosstab(
    df_clean1['d02o1b'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d02o1b,,,,,,,,,,
No,385,413,376,414,331,392,329,352,5,1
No sé,1,6,1,2,1,1,0,0,0,0
Sí,1,4,1,5,3,4,4,8,0,0


In [1113]:
df_clean1['d02o1c'].value_counts(dropna=False).sort_index()

,count
d02o1c,
No,2987
No sé,23
Sí,30


In [1114]:
pd.crosstab(
    df_clean1['d02o1c'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d02o1c,,,,
No,517,1,45,2424
No sé,5,0,0,18
Sí,10,0,1,19


In [1115]:
pd.crosstab(
    df_clean1['d02o1c'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d02o1c,,,,,,,,,,
No,382,418,369,414,327,388,328,355,5,1
No sé,4,5,7,2,1,3,1,0,0,0
Sí,1,0,2,5,7,6,4,5,0,0


In [1116]:
df_clean1['d02o1d'].value_counts(dropna=False).sort_index()

,count
d02o1d,
No,2984
No sé,21
Sí,35


In [1117]:
pd.crosstab(
    df_clean1['d02o1d'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d02o1d,,,,
No,517,1,46,2420
No sé,5,0,0,16
Sí,10,0,0,25


In [1118]:
pd.crosstab(
    df_clean1['d02o1d'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d02o1d,,,,,,,,,,
No,382,417,372,411,330,386,328,352,5,1
No sé,3,5,2,4,1,5,1,0,0,0
Sí,2,1,4,6,4,6,4,8,0,0


In [1119]:
df_final['Diabetes'] = (df_clean1['d02o1a'] == 'Sí').astype(int)

df_final['Diabetes'].value_counts(dropna=False)

,count
Diabetes,
0,3024
1,16


In [1120]:
df_final['Hipertensión'] = (df_clean1['d02o1b'] == 'Sí').astype(int)

df_final['Hipertensión'].value_counts(dropna=False)

,count
Hipertensión,
0,3010
1,30


In [1121]:
df_final['Colesterol'] = (df_clean1['d02o1c'] == 'Sí').astype(int)

df_final['Colesterol'].value_counts(dropna=False)

,count
Colesterol,
0,3010
1,30


In [1122]:
df_final['Triglicéridos'] = (df_clean1['d02o1d'] == 'Sí').astype(int)

df_final['Triglicéridos'].value_counts(dropna=False)

,count
Triglicéridos,
0,3005
1,35


In [1123]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,Hipertensión,Colesterol,Triglicéridos
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,0,0,0
2,0,13,0,0,0,0,0,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0
4,0,13,1,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0


### Variable de Embarazo

Recordemos como se distribuye esta variable. Aquí se puede observar que, como era de esperarse, solo a las mujeres se les hizo esta pregunta. Dada su relevancia, se especificará otro modelo. Dado que ésta es la segunda partición, uno puede esperar que se necesiten 4 modelos, sin embargo, como se puede observar en los datos, solo adolescentes mayores de 13 años han reportado estar embarazada por que se hara solamente una división de genero para la población de 12 años en adelante

Respecto al hombre con respuesta de "No", se asume que es un error de captura de datos por lo que se decide prescindir de esta observación (se hará al final de construir df_final)

In [1124]:
df_clean1['d0313'].value_counts(dropna=False).sort_index()

,count
d0313,
No,72
Sí,37
NaN,2931


In [1125]:
pd.crosstab(
    df_clean1['d0313'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0313,,,,
No,8,0,2,62
Sí,5,0,0,32
NaN,519,1,44,2367


In [1126]:
pd.crosstab(
    df_clean1['d0313'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0313,,,,,,,,,,
No,0,0,0,1,3,14,23,31,0,0
Sí,0,0,0,1,1,5,11,19,0,0
NaN,387,423,378,419,331,378,299,310,5,1


In [1127]:
pd.crosstab(
    df_clean1['d0313'],
    df_clean1['sexo'],
    dropna=False
)

sexo,Hombre,Mujer
d0313,,
No,1,71
Sí,0,37
NaN,1540,1391


In [1128]:
#df_final['Embarazo'] = (df_clean1['d0313'] == 'Sí').astype(int)

df_final['Embarazo'] = np.where(
    (df_clean1['d0313'].isna()) & (df_clean1['sexo'] == 'Hombre'),
    pd.NA,
    (df_clean1['d0313'] == 'Sí').astype(int)
)

df_final['Embarazo'].value_counts(dropna=False)

,count
Embarazo,
<NA>,1540
0,1463
1,37


In [1129]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,Hipertensión,Colesterol,Triglicéridos,Embarazo
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,0,0,0,<NA>
2,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0
4,0,13,1,0,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>
3036,0,13,1,0,0,0,0,0,0,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0


### Variables de Salud Mental

Recordemos como se distribuyen estas variables. Dada la similitud entre varias de estas preguntas las colapsaremos en menos columnas

* d0421 (Ansiedad) --> En caso de diaramente y semanalmente se pondrá un 1, de lo contrario un cero

* d0422, d0601a, d0601c, d0601g (Depresión) --> d0422 en caso de diaramente y semanalmente se pondrá un 1, de lo contrario un cero. Para las respuestas en d0601X se considerará como 1 aquellas respuestas con "Un número de veces considerables (3-4 días)" o "Todo el tiempo o la mayoria del tiempo (5-7 días)"

* d0601b (Dificultad de concentración) --> Para las respuestas en d0601X se considerará como 1 aquellas respuestas con "Un número de veces considerables (3-4 días)" o "Todo el tiempo o la mayoria del tiempo (5-7 días)"

* d0601e (Dormir mal) --> Similar al punto anterior

* d0817 (Pensamiento suicida) --> Sí será igual a 1 y las demas respuestas serán 0

* Se decide ignorar la columna d0601d por amibgua, la columna d0601f ya que es el inverso de la pregunta de depresión y la columna d0818 ya que agrega magnitud a la columna d0817 pero estamos manteniendo todas nuestras variables como dicotómicas

In [1130]:
df_clean1['d0421'].value_counts(dropna=False).sort_index()

,count
d0421,
Diariamente,243
Mensualmente,467
Nunca,1063
Semanalmente,511
Varias veces al año,756


In [1131]:
pd.crosstab(
    df_clean1['d0421'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0421,,,,
Diariamente,58,0,7,178
Mensualmente,78,0,6,383
Nunca,145,1,13,904
Semanalmente,111,0,15,385
Varias veces al año,140,0,5,611


In [1132]:
pd.crosstab(
    df_clean1['d0421'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0421,,,,,,,,,,
Diariamente,28,30,20,39,24,40,29,33,0,0
Mensualmente,45,49,51,57,66,60,52,86,1,0
Nunca,158,178,150,156,101,118,106,92,3,1
Semanalmente,54,51,60,71,66,83,63,63,0,0
Varias veces al año,102,115,97,98,78,96,83,86,1,0


In [1133]:
df_clean1['d0422'].value_counts(dropna=False).sort_index()

,count
d0422,
Diariamente,95
Mensualmente,425
Nunca,1389
Semanalmente,345
Varias veces al año,786


In [1134]:
pd.crosstab(
    df_clean1['d0422'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0422,,,,
Diariamente,25,0,5,65
Mensualmente,80,0,7,338
Nunca,193,1,15,1180
Semanalmente,95,0,10,240
Varias veces al año,139,0,9,638


In [1135]:
pd.crosstab(
    df_clean1['d0422'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0422,,,,,,,,,,
Diariamente,14,9,13,10,11,17,8,13,0,0
Mensualmente,41,44,47,58,57,57,50,71,0,0
Nunca,186,228,186,211,131,165,140,138,3,1
Semanalmente,39,36,42,37,57,60,38,36,0,0
Varias veces al año,107,106,90,105,79,98,97,102,2,0


In [1136]:
df_clean1['d0601a'].value_counts(dropna=False).sort_index()

,count
d0601a,
Pocas veces o alguna vez (1-2 días),378
Rara vez o nunca (menos de un día),2487
Todo el tiempo o la mayoria del tiempo (5-7 días),62
Un número de veces considerables (3-4 días),113


In [1137]:
pd.crosstab(
    df_clean1['d0601a'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601a,,,,
Pocas veces o alguna vez (1-2 días),104,0,12,262
Rara vez o nunca (menos de un día),377,1,22,2087
Todo el tiempo o la mayoria del tiempo (5-7 días),14,0,5,43
Un número de veces considerables (3-4 días),37,0,7,69


In [1138]:
pd.crosstab(
    df_clean1['d0601a'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601a,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),50,57,38,41,53,61,36,42,0,0
Rara vez o nunca (menos de un día),319,352,323,352,259,313,274,289,5,1
Todo el tiempo o la mayoria del tiempo (5-7 días),4,6,6,12,9,5,8,12,0,0
Un número de veces considerables (3-4 días),14,8,11,16,14,18,15,17,0,0


In [1139]:
df_clean1['d0601b'].value_counts(dropna=False).sort_index()

,count
d0601b,
Pocas veces o alguna vez (1-2 días),724
Rara vez o nunca (menos de un día),2050
Todo el tiempo o la mayoria del tiempo (5-7 días),81
Un número de veces considerables (3-4 días),185


In [1140]:
pd.crosstab(
    df_clean1['d0601b'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601b,,,,
Pocas veces o alguna vez (1-2 días),188,1,10,525
Rara vez o nunca (menos de un día),248,0,14,1788
Todo el tiempo o la mayoria del tiempo (5-7 días),31,0,12,38
Un número de veces considerables (3-4 días),65,0,10,110


In [1141]:
pd.crosstab(
    df_clean1['d0601b'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601b,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),93,102,76,95,101,92,75,90,0,0
Rara vez o nunca (menos de un día),260,303,279,280,202,260,228,232,5,1
Todo el tiempo o la mayoria del tiempo (5-7 días),9,2,10,17,10,11,6,16,0,0
Un número de veces considerables (3-4 días),25,16,13,29,22,34,24,22,0,0


In [1142]:
df_clean1['d0601c'].value_counts(dropna=False).sort_index()

,count
d0601c,
Pocas veces o alguna vez (1-2 días),437
Rara vez o nunca (menos de un día),2437
Todo el tiempo o la mayoria del tiempo (5-7 días),58
Un número de veces considerables (3-4 días),108


In [1143]:
pd.crosstab(
    df_clean1['d0601c'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601c,,,,
Pocas veces o alguna vez (1-2 días),116,0,13,308
Rara vez o nunca (menos de un día),370,0,23,2044
Todo el tiempo o la mayoria del tiempo (5-7 días),16,0,3,39
Un número de veces considerables (3-4 días),30,1,7,70


In [1144]:
pd.crosstab(
    df_clean1['d0601c'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601c,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),56,52,44,46,64,63,47,65,0,0
Rara vez o nunca (menos de un día),317,361,318,352,246,307,262,268,5,1
Todo el tiempo o la mayoria del tiempo (5-7 días),4,4,5,9,10,9,8,9,0,0
Un número de veces considerables (3-4 días),10,6,11,14,15,18,16,18,0,0


In [1145]:
df_clean1['d0601d'].value_counts(dropna=False).sort_index()

,count
d0601d,
Pocas veces o alguna vez (1-2 días),582
Rara vez o nunca (menos de un día),2141
Todo el tiempo o la mayoria del tiempo (5-7 días),145
Un número de veces considerables (3-4 días),172


In [1146]:
pd.crosstab(
    df_clean1['d0601d'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601d,,,,
Pocas veces o alguna vez (1-2 días),152,1,13,416
Rara vez o nunca (menos de un día),292,0,19,1830
Todo el tiempo o la mayoria del tiempo (5-7 días),38,0,7,100
Un número de veces considerables (3-4 días),50,0,7,115


In [1147]:
pd.crosstab(
    df_clean1['d0601d'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601d,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),80,76,73,76,79,80,59,58,1,0
Rara vez o nunca (menos de un día),268,316,272,302,222,271,229,256,4,1
Todo el tiempo o la mayoria del tiempo (5-7 días),19,13,13,22,18,19,17,24,0,0
Un número de veces considerables (3-4 días),20,18,20,21,16,27,28,22,0,0


In [1148]:
df_clean1['d0601e'].value_counts(dropna=False).sort_index()

,count
d0601e,
Pocas veces o alguna vez (1-2 días),429
Rara vez o nunca (menos de un día),2009
Todo el tiempo o la mayoria del tiempo (5-7 días),413
Un número de veces considerables (3-4 días),189


In [1149]:
pd.crosstab(
    df_clean1['d0601e'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601e,,,,
Pocas veces o alguna vez (1-2 días),100,1,11,317
Rara vez o nunca (menos de un día),320,0,21,1668
Todo el tiempo o la mayoria del tiempo (5-7 días),65,0,8,340
Un número de veces considerables (3-4 días),47,0,6,136


In [1150]:
pd.crosstab(
    df_clean1['d0601e'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601e,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),40,49,49,55,51,68,51,66,0,0
Rara vez o nunca (menos de un día),276,319,254,293,212,237,214,199,5,0
Todo el tiempo o la mayoria del tiempo (5-7 días),55,45,56,50,51,50,46,59,0,1
Un número de veces considerables (3-4 días),16,10,19,23,21,42,22,36,0,0


In [1151]:
df_clean1['d0601f'].value_counts(dropna=False).sort_index()

,count
d0601f,
Pocas veces o alguna vez (1-2 días),128
Rara vez o nunca (menos de un día),321
Todo el tiempo o la mayoria del tiempo (5-7 días),2324
Un número de veces considerables (3-4 días),267


In [1152]:
pd.crosstab(
    df_clean1['d0601f'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601f,,,,
Pocas veces o alguna vez (1-2 días),35,0,4,89
Rara vez o nunca (menos de un día),34,0,2,285
Todo el tiempo o la mayoria del tiempo (5-7 días),393,1,31,1899
Un número de veces considerables (3-4 días),70,0,9,188


In [1153]:
pd.crosstab(
    df_clean1['d0601f'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601f,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),7,13,13,16,13,25,15,26,0,0
Rara vez o nunca (menos de un día),37,55,43,41,35,42,38,30,0,0
Todo el tiempo o la mayoria del tiempo (5-7 días),324,330,287,332,249,296,238,262,5,1
Un número de veces considerables (3-4 días),19,25,35,32,38,34,42,42,0,0


In [1154]:
df_clean1['d0601g'].value_counts(dropna=False).sort_index()

,count
d0601g,
Pocas veces o alguna vez (1-2 días),484
Rara vez o nunca (menos de un día),2360
Todo el tiempo o la mayoria del tiempo (5-7 días),63
Un número de veces considerables (3-4 días),133


In [1155]:
pd.crosstab(
    df_clean1['d0601g'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0601g,,,,
Pocas veces o alguna vez (1-2 días),128,1,11,344
Rara vez o nunca (menos de un día),345,0,26,1989
Todo el tiempo o la mayoria del tiempo (5-7 días),18,0,1,44
Un número de veces considerables (3-4 días),41,0,8,84


In [1156]:
pd.crosstab(
    df_clean1['d0601g'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0601g,,,,,,,,,,
Pocas veces o alguna vez (1-2 días),65,65,58,61,60,69,47,58,1,0
Rara vez o nunca (menos de un día),300,343,298,335,249,297,254,279,4,1
Todo el tiempo o la mayoria del tiempo (5-7 días),6,5,6,9,9,6,11,11,0,0
Un número de veces considerables (3-4 días),16,10,16,16,17,25,21,12,0,0


In [1157]:
df_clean1['d0817'].value_counts(dropna=False).sort_index()

,count
d0817,
No,2826
No responde,47
Sí,167


In [1158]:
pd.crosstab(
    df_clean1['d0817'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0817,,,,
No,477,1,38,2310
No responde,9,0,2,36
Sí,46,0,6,115


In [1159]:
pd.crosstab(
    df_clean1['d0817'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0817,,,,,,,,,,
No,380,410,360,391,300,365,300,315,4,1
No responde,2,5,4,7,6,6,8,9,0,0
Sí,5,8,14,23,29,26,25,36,1,0


In [1160]:
df_clean1['d0818'].value_counts(dropna=False).sort_index()

,count
d0818,
Algunas veces,83
Casi siempre,9
No responde,2
Rara vez,61
Siempre o muy frecuente,12
NaN,2873


In [1161]:
pd.crosstab(
    df_clean1['d0818'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0818,,,,
Algunas veces,29,0,4,50
Casi siempre,4,0,0,5
No responde,0,0,0,2
Rara vez,11,0,1,49
Siempre o muy frecuente,2,0,1,9
NaN,486,1,40,2346


In [1162]:
pd.crosstab(
    df_clean1['d0818'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0818,,,,,,,,,,
Algunas veces,3,2,9,12,17,8,14,17,1,0
Casi siempre,0,3,0,1,1,2,0,2,0,0
No responde,0,0,0,0,0,1,1,0,0,0
Rara vez,1,2,3,10,7,13,9,16,0,0
Siempre o muy frecuente,1,1,2,0,4,2,1,1,0,0
NaN,382,415,364,398,306,371,308,324,4,1


In [1163]:
df_final['Ansiedad'] = ((df_clean1['d0421'] == 'Diariamente') | (df_clean1['d0421'] == 'Semanalmente')).astype(int)

df_final['Ansiedad'].value_counts(dropna=False)

,count
Ansiedad,
0,2286
1,754


In [1164]:
df_final['Depresión'] = ((df_clean1['d0422'] == 'Diariamente') | (df_clean1['d0422'] == 'Semanalmente') |
                         (df_clean1['d0601a'] == 'Un número de veces considerables (3-4 días)') | (df_clean1['d0601a'] == 'Todo el tiempo o la mayoria del tiempo (5-7 días)') |
                         (df_clean1['d0601c'] == 'Un número de veces considerables (3-4 días)') | (df_clean1['d0601c'] == 'Todo el tiempo o la mayoria del tiempo (5-7 días)') |
                         (df_clean1['d0601g'] == 'Un número de veces considerables (3-4 días)') | (df_clean1['d0601g'] == 'Todo el tiempo o la mayoria del tiempo (5-7 días)')).astype(int)

df_final['Depresión'].value_counts(dropna=False)

,count
Depresión,
0,2414
1,626


In [1165]:
df_final['Dificultad de concentración'] = ((df_clean1['d0601b'] == 'Un número de veces considerables (3-4 días)') |
                                           (df_clean1['d0601b'] == 'Todo el tiempo o la mayoria del tiempo (5-7 días)')).astype(int)

df_final['Dificultad de concentración'].value_counts(dropna=False)

,count
Dificultad de concentración,
0,2774
1,266


In [1166]:
df_final['Dormir mal'] = ((df_clean1['d0601e'] == 'Un número de veces considerables (3-4 días)') |
                          (df_clean1['d0601e'] == 'Todo el tiempo o la mayoria del tiempo (5-7 días)')).astype(int)

df_final['Dormir mal'].value_counts(dropna=False)

,count
Dormir mal,
0,2438
1,602


In [1167]:
df_final['Pensamiento suicida'] = (df_clean1['d0817'] == 'Sí').astype(int)

df_final['Pensamiento suicida'].value_counts(dropna=False)

,count
Pensamiento suicida,
0,2873
1,167


In [1168]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,Hipertensión,Colesterol,Triglicéridos,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,1,0,0,0
1,0,15,0,0,1,0,0,0,0,0,0,0,0,<NA>,0,0,0,1,1
2,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,1,0
4,0,13,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,0,0


### Variables de Conductas Alimentarias

Recordemos como se distribuyen estas variables. Se puede observar que la primera pregunta (frecuencia por preocuparse por engordar) es donde hay más personas en el grupo de muy frecuentemente. Dado que todas las preguntas están relacionadas con trastornos alimenticios (eso sí, de diferente severidad) se van a colapsar todas las preguntas a una sola variable predictora llamada "Desordenes alimenticios" en donde se considerarán exclusivamente personas que hayan respondido "Muy frecuentemente (más de 2 veces en una semana)" a alguna de las preguntas

In [1169]:
df_clean1['d06a1'].value_counts(dropna=False).sort_index()

,count
d06a1,
Algunas veces,670
Frecuentemente (2 veces en una semana),199
Muy frecuentemente (más de 2 veces en una semana),129
Nunca o casi nunca,2042


In [1170]:
pd.crosstab(
    df_clean1['d06a1'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a1,,,,
Algunas veces,140,0,9,521
Frecuentemente (2 veces en una semana),49,0,4,146
Muy frecuentemente (más de 2 veces en una semana),32,0,4,93
Nunca o casi nunca,311,1,29,1701


In [1171]:
pd.crosstab(
    df_clean1['d06a1'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a1,,,,,,,,,,
Algunas veces,86,107,68,90,76,88,85,68,1,1
Frecuentemente (2 veces en una semana),11,20,22,29,26,37,21,33,0,0
Muy frecuentemente (más de 2 veces en una semana),7,7,14,17,19,19,20,26,0,0
Nunca o casi nunca,283,289,274,285,214,253,207,233,4,0


In [1172]:
df_clean1['d06a2'].value_counts(dropna=False).sort_index()

,count
d06a2,
Algunas veces,1121
Frecuentemente (2 veces en una semana),214
Muy frecuentemente (más de 2 veces en una semana),81
Nunca o casi nunca,1624


In [1173]:
pd.crosstab(
    df_clean1['d06a2'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a2,,,,
Algunas veces,242,0,16,863
Frecuentemente (2 veces en una semana),51,0,3,160
Muy frecuentemente (más de 2 veces en una semana),22,0,3,56
Nunca o casi nunca,217,1,24,1382


In [1174]:
pd.crosstab(
    df_clean1['d06a2'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a2,,,,,,,,,,
Algunas veces,134,156,140,180,114,147,130,118,2,0
Frecuentemente (2 veces en una semana),16,19,20,30,21,33,34,41,0,0
Muy frecuentemente (más de 2 veces en una semana),3,6,8,12,9,19,12,12,0,0
Nunca o casi nunca,234,242,210,199,191,198,157,189,3,1


In [1175]:
df_clean1['d06a3'].value_counts(dropna=False).sort_index()

,count
d06a3,
Algunas veces,529
Frecuentemente (2 veces en una semana),105
Muy frecuentemente (más de 2 veces en una semana),41
Nunca o casi nunca,2365


In [1176]:
pd.crosstab(
    df_clean1['d06a3'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a3,,,,
Algunas veces,142,0,9,378
Frecuentemente (2 veces en una semana),29,0,2,74
Muy frecuentemente (más de 2 veces en una semana),12,0,3,26
Nunca o casi nunca,349,1,32,1983


In [1177]:
pd.crosstab(
    df_clean1['d06a3'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a3,,,,,,,,,,
Algunas veces,66,63,62,72,61,75,72,58,0,0
Frecuentemente (2 veces en una semana),10,9,10,19,7,12,18,20,0,0
Muy frecuentemente (más de 2 veces en una semana),4,1,3,8,3,10,4,8,0,0
Nunca o casi nunca,307,350,303,322,264,300,239,274,5,1


In [1178]:
df_clean1['d06a4'].value_counts(dropna=False).sort_index()

,count
d06a4,
Algunas veces,91
Frecuentemente (2 veces en una semana),12
Muy frecuentemente (más de 2 veces en una semana),5
Nunca o casi nunca,2932


In [1179]:
pd.crosstab(
    df_clean1['d06a4'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a4,,,,
Algunas veces,20,0,3,68
Frecuentemente (2 veces en una semana),3,0,1,8
Muy frecuentemente (más de 2 veces en una semana),0,0,1,4
Nunca o casi nunca,509,1,41,2381


In [1180]:
pd.crosstab(
    df_clean1['d06a4'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a4,,,,,,,,,,
Algunas veces,12,12,7,12,12,16,12,8,0,0
Frecuentemente (2 veces en una semana),1,1,0,0,2,1,4,3,0,0
Muy frecuentemente (más de 2 veces en una semana),0,0,1,0,0,1,3,0,0,0
Nunca o casi nunca,374,410,370,409,321,379,314,349,5,1


In [1181]:
df_clean1['d06a5'].value_counts(dropna=False).sort_index()

,count
d06a5,
Algunas veces,180
Frecuentemente (2 veces en una semana),33
Muy frecuentemente (más de 2 veces en una semana),24
Nunca o casi nunca,2803


In [1182]:
pd.crosstab(
    df_clean1['d06a5'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a5,,,,
Algunas veces,46,0,3,131
Frecuentemente (2 veces en una semana),9,0,1,23
Muy frecuentemente (más de 2 veces en una semana),4,0,2,18
Nunca o casi nunca,473,1,40,2289


In [1183]:
pd.crosstab(
    df_clean1['d06a5'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a5,,,,,,,,,,
Algunas veces,16,13,16,26,30,25,32,21,0,1
Frecuentemente (2 veces en una semana),2,3,1,4,4,8,2,9,0,0
Muy frecuentemente (más de 2 veces en una semana),1,3,2,2,4,4,4,4,0,0
Nunca o casi nunca,368,404,359,389,297,360,295,326,5,0


In [1184]:
df_clean1['d06a6'].value_counts(dropna=False).sort_index()

,count
d06a6,
Algunas veces,252
Frecuentemente (2 veces en una semana),49
Muy frecuentemente (más de 2 veces en una semana),17
Nunca o casi nunca,2722


In [1185]:
pd.crosstab(
    df_clean1['d06a6'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a6,,,,
Algunas veces,65,0,10,177
Frecuentemente (2 veces en una semana),12,0,0,37
Muy frecuentemente (más de 2 veces en una semana),2,0,1,14
Nunca o casi nunca,453,1,35,2233


In [1186]:
pd.crosstab(
    df_clean1['d06a6'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a6,,,,,,,,,,
Algunas veces,21,29,27,42,35,36,35,26,0,1
Frecuentemente (2 veces en una semana),4,7,3,5,3,11,6,10,0,0
Muy frecuentemente (más de 2 veces en una semana),1,4,1,0,3,2,4,2,0,0
Nunca o casi nunca,361,383,347,374,294,348,288,322,5,0


In [1187]:
df_clean1['d06a7'].value_counts(dropna=False).sort_index()

,count
d06a7,
Algunas veces,273
Frecuentemente (2 veces en una semana),73
Muy frecuentemente (más de 2 veces en una semana),35
Nunca o casi nunca,2659


In [1188]:
pd.crosstab(
    df_clean1['d06a7'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a7,,,,
Algunas veces,72,0,10,191
Frecuentemente (2 veces en una semana),15,0,1,57
Muy frecuentemente (más de 2 veces en una semana),9,0,2,24
Nunca o casi nunca,436,1,33,2189


In [1189]:
pd.crosstab(
    df_clean1['d06a7'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a7,,,,,,,,,,
Algunas veces,30,39,28,43,33,34,33,33,0,0
Frecuentemente (2 veces en una semana),3,7,8,15,18,8,6,8,0,0
Muy frecuentemente (más de 2 veces en una semana),2,6,2,4,5,7,5,4,0,0
Nunca o casi nunca,352,371,340,359,279,348,289,315,5,1


In [1190]:
df_clean1['d06a8'].value_counts(dropna=False).sort_index()

,count
d06a8,
Algunas veces,43
Frecuentemente (2 veces en una semana),6
Muy frecuentemente (más de 2 veces en una semana),6
Nunca o casi nunca,2985


In [1191]:
pd.crosstab(
    df_clean1['d06a8'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a8,,,,
Algunas veces,11,0,1,31
Frecuentemente (2 veces en una semana),3,0,0,3
Muy frecuentemente (más de 2 veces en una semana),0,0,1,5
Nunca o casi nunca,518,1,44,2422


In [1192]:
pd.crosstab(
    df_clean1['d06a8'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a8,,,,,,,,,,
Algunas veces,6,6,8,5,3,8,4,3,0,0
Frecuentemente (2 veces en una semana),1,0,1,3,0,0,0,1,0,0
Muy frecuentemente (más de 2 veces en una semana),0,1,1,0,0,1,1,2,0,0
Nunca o casi nunca,380,416,368,413,332,388,328,354,5,1


In [1193]:
df_clean1['d06a9'].value_counts(dropna=False).sort_index()

,count
d06a9,
Algunas veces,43
Frecuentemente (2 veces en una semana),2
Muy frecuentemente (más de 2 veces en una semana),8
Nunca o casi nunca,2987


In [1194]:
pd.crosstab(
    df_clean1['d06a9'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a9,,,,
Algunas veces,9,0,1,33
Frecuentemente (2 veces en una semana),0,0,0,2
Muy frecuentemente (más de 2 veces en una semana),1,0,2,5
Nunca o casi nunca,522,1,43,2421


In [1195]:
pd.crosstab(
    df_clean1['d06a9'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a9,,,,,,,,,,
Algunas veces,7,6,6,6,4,5,4,5,0,0
Frecuentemente (2 veces en una semana),0,0,0,0,1,0,0,1,0,0
Muy frecuentemente (más de 2 veces en una semana),1,2,1,2,0,1,1,0,0,0
Nunca o casi nunca,379,415,371,413,330,391,328,354,5,1


In [1196]:
df_clean1['d06a10'].value_counts(dropna=False).sort_index()

,count
d06a10,
Algunas veces,41
Frecuentemente (2 veces en una semana),3
Muy frecuentemente (más de 2 veces en una semana),4
Nunca o casi nunca,2992


In [1197]:
pd.crosstab(
    df_clean1['d06a10'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d06a10,,,,
Algunas veces,8,0,1,32
Frecuentemente (2 veces en una semana),1,0,0,2
Muy frecuentemente (más de 2 veces en una semana),0,0,1,3
Nunca o casi nunca,523,1,44,2424


In [1198]:
pd.crosstab(
    df_clean1['d06a10'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d06a10,,,,,,,,,,
Algunas veces,8,5,3,3,4,6,5,7,0,0
Frecuentemente (2 veces en una semana),0,0,0,1,1,0,0,1,0,0
Muy frecuentemente (más de 2 veces en una semana),0,0,1,0,0,1,1,1,0,0
Nunca o casi nunca,379,418,374,417,330,390,327,351,5,1


In [1199]:
df_final['Desordenes alimenticios'] = ((df_clean1['d06a1'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a2'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a3'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a4'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a5'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a6'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a7'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a8'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a9'] == 'Muy frecuentemente (más de 2 veces en una semana)') |
                                       (df_clean1['d06a10'] == 'Muy frecuentemente (más de 2 veces en una semana)')).astype(int)

df_final['Desordenes alimenticios'].value_counts(dropna=False)

,count
Desordenes alimenticios,
0,2805
1,235


In [1200]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,Hipertensión,Colesterol,Triglicéridos,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,1,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,0,0,0,<NA>,0,0,0,1,1,0
2,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,1,0,0
4,0,13,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,0,0,0,<NA>,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,0,0,0,0,0,0,0,0,0,0


### Variables de Accidentes

Recordemos como se distribuyen estas variables. Dado que d0702 y d0709 agregan detalle a la pregunta de si la persona se ha accidentado, se decide no utilizar esas columans y quedarnos exclusivamente con la variable de si la persona se ha accidentado o no en los últimos 12 meses

In [1201]:
df_clean1['d0701'].value_counts(dropna=False).sort_index()

,count
d0701,
No,2835
No sabe,4
Sí,201


In [1202]:
pd.crosstab(
    df_clean1['d0701'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0701,,,,
No,486,1,42,2306
No sabe,1,0,0,3
Sí,45,0,4,152


In [1203]:
pd.crosstab(
    df_clean1['d0701'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0701,,,,,,,,,,
No,350,395,353,400,313,373,306,340,4,1
No sabe,0,1,2,0,0,0,1,0,0,0
Sí,37,27,23,21,22,24,26,20,1,0


In [1204]:
df_clean1['d0702'].value_counts(dropna=False).sort_index()

,count
d0702,
Atropellado como peatón,2
Caída a nivel de piso,101
"Caída de un nivel a otro: silla, escalera, etc.",21
Choque en motocicleta,21
Choque entre vehículos de 4 o más ruedas,7
Choque o atropellamiento como ciclista,5
Contacto con superficies o sustancias calientes (quemaduras),1
Exposición a humo o fuego,3
Exposición accidental a otros factores,4


In [1205]:
pd.crosstab(
    df_clean1['d0702'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0702,,,,
Atropellado como peatón,2,0,0,0
Caída a nivel de piso,19,0,2,80
"Caída de un nivel a otro: silla, escalera, etc.",5,0,0,16
Choque en motocicleta,4,0,0,17
Choque entre vehículos de 4 o más ruedas,2,0,0,5
Choque o atropellamiento como ciclista,0,0,0,5
Contacto con superficies o sustancias calientes (quemaduras),0,0,0,1
Exposición a humo o fuego,2,0,0,1
Exposición accidental a otros factores,1,0,0,3


In [1206]:
pd.crosstab(
    df_clean1['d0702'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0702,,,,,,,,,,
Atropellado como peatón,0,0,0,2,0,0,0,0,0,0
Caída a nivel de piso,19,18,12,15,13,9,10,5,0,0
"Caída de un nivel a otro: silla, escalera, etc.",2,5,3,1,4,1,5,0,0,0
Choque en motocicleta,2,1,1,0,3,4,4,5,1,0
Choque entre vehículos de 4 o más ruedas,0,0,1,0,0,2,3,1,0,0
Choque o atropellamiento como ciclista,2,1,0,0,0,1,0,1,0,0
Contacto con superficies o sustancias calientes (quemaduras),0,0,1,0,0,0,0,0,0,0
Exposición a humo o fuego,1,0,1,0,0,0,0,1,0,0
Exposición accidental a otros factores,0,1,0,1,1,1,0,0,0,0


In [1207]:
df_clean1['d0709'].value_counts(dropna=False).sort_index()

,count
d0709,
Dificultad para ver o ceguera,1
Limitación o dificultad para moverse o caminar (o lo hace con ayuda),13
Limitación o dificultad para usar sus brazos y manos,5
NS/NR,2
Ningún problema,173
Otra limitación física o mental,4
NaN,2842


In [1208]:
pd.crosstab(
    df_clean1['d0709'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0709,,,,
Dificultad para ver o ceguera,0,0,0,1
Limitación o dificultad para moverse o caminar (o lo hace con ayuda),3,0,0,10
Limitación o dificultad para usar sus brazos y manos,1,0,0,4
NS/NR,0,0,0,2
Ningún problema,41,0,3,129
Otra limitación física o mental,0,0,1,3
NaN,487,1,42,2312


In [1209]:
pd.crosstab(
    df_clean1['d0709'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0709,,,,,,,,,,
Dificultad para ver o ceguera,1,0,0,0,0,0,0,0,0,0
Limitación o dificultad para moverse o caminar (o lo hace con ayuda),1,2,1,2,1,1,2,3,0,0
Limitación o dificultad para usar sus brazos y manos,1,0,0,1,2,0,1,0,0,0
NS/NR,0,1,1,0,0,0,0,0,0,0
Ningún problema,33,23,21,18,17,21,22,17,1,0
Otra limitación física o mental,1,1,0,0,1,0,1,0,0,0
NaN,350,396,355,400,314,375,307,340,4,1


In [1210]:
df_final['Accidente'] = (df_clean1['d0701'] == 'Sí').astype(int)

df_final['Accidente'].value_counts(dropna=False)

,count
Accidente,
0,2839
1,201


In [1211]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Colesterol,Triglicéridos,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,1,0,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,...,0,0,<NA>,0,0,0,1,1,0,0
2,0,13,0,0,0,0,0,0,0,0,...,0,0,<NA>,0,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,0,0,1,0,0,0
4,0,13,1,0,0,0,0,0,0,0,...,0,0,0,0,1,1,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,...,0,0,<NA>,0,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,...,0,0,0,1,0,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,0,0,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,0,0,0,0,0,0


### Variables de Coerción o Violencia

Recordemos como se distribuyen estas variables. Para la primera que evalua si la primera relación sexual fue forzada vemos, afortunadamente, muy pocos casos por lo que se decide no utilizar esta variable. Para las otras variables si vemos mayor volumen por lo que se decide quedarse con dichas variables. Para la última (Acoso sexual) se considerará como 1 tanto el acoso previo a los 12 años de edad como el acoso posterior a los 12 años de edad

In [1212]:
df_clean1['d0309'].value_counts(dropna=False).sort_index()

,count
d0309,
No,233
No recuerdo,5
Sí,7
NaN,2795


In [1213]:
pd.crosstab(
    df_clean1['d0309'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0309,,,,
No,29,0,3,201
No recuerdo,2,0,0,3
Sí,1,0,0,6
NaN,500,1,43,2251


In [1214]:
pd.crosstab(
    df_clean1['d0309'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0309,,,,,,,,,,
No,0,0,4,5,6,41,73,104,0,0
No recuerdo,0,0,0,0,1,1,2,1,0,0
Sí,0,0,0,0,0,2,2,3,0,0
NaN,387,423,374,416,328,353,256,252,5,1


In [1215]:
df_clean1['d0801'].value_counts(dropna=False).sort_index()

,count
d0801,
No,2986
No responde,2
Sí,52


In [1216]:
pd.crosstab(
    df_clean1['d0801'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0801,,,,
No,511,1,45,2429
No responde,1,0,0,1
Sí,20,0,1,31


In [1217]:
pd.crosstab(
    df_clean1['d0801'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0801,,,,,,,,,,
No,380,414,376,418,327,390,323,353,4,1
No responde,1,0,0,0,0,0,0,1,0,0
Sí,6,9,2,3,8,7,10,6,1,0


In [1218]:
df_clean1['d0810'].value_counts(dropna=False).sort_index()

,count
d0810,
No recuerda,36
No responde,15
"No, nunca",2882
"Sí, antes de los 12",84
"Sí, cuando tenía 12 o más años",23


In [1219]:
pd.crosstab(
    df_clean1['d0810'],
    df_clean1['d0415'],
    dropna=False
)

d0415,Cierta dificultad,Le resulta imposible aprender,Mucha dificultad,Ninguna dificultad
d0810,,,,
No recuerda,9,0,0,27
No responde,3,0,1,11
"No, nunca",491,1,39,2351
"Sí, antes de los 12",21,0,6,57
"Sí, cuando tenía 12 o más años",8,0,0,15


In [1220]:
pd.crosstab(
    df_clean1['d0810'],
    df_clean1['edad'],
    dropna=False
)

edad,10,11,12,13,14,15,16,17,18,19
d0810,,,,,,,,,,
No recuerda,6,6,1,8,1,7,4,3,0,0
No responde,2,2,0,1,0,2,5,3,0,0
"No, nunca",376,407,368,394,319,370,310,332,5,1
"Sí, antes de los 12",3,8,9,14,10,14,11,15,0,0
"Sí, cuando tenía 12 o más años",0,0,0,4,5,4,3,7,0,0


In [1222]:
df_final['Robo Agresión o Violencia'] = (df_clean1['d0801'] == 'Sí').astype(int)

df_final['Robo Agresión o Violencia'].value_counts(dropna=False)

,count
Robo Agresión o Violencia,
0,2988
1,52


In [1223]:
df_final['Acoso Sexual'] = ((df_clean1['d0810'] == 'Sí, antes de los 12') | (df_clean1['d0810'] == 'Sí, cuando tenía 12 o más años')).astype(int)

df_final['Acoso Sexual'].value_counts(dropna=False)

,count
Acoso Sexual,
0,2933
1,107


In [1224]:
df_final

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente,Robo Agresión o Violencia,Acoso Sexual
0,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,1,0,0,0,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,...,<NA>,0,0,0,1,1,0,0,0,0
2,0,13,0,0,0,0,0,0,0,0,...,<NA>,0,0,0,0,0,0,0,0,0
3,0,10,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,1,0,0,0,0,0
4,0,13,1,0,0,0,0,0,0,0,...,0,0,1,1,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,...,<NA>,0,0,0,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,0
3037,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,0,0,0,0,0,0
3038,0,11,1,0,0,<NA>,<NA>,<NA>,<NA>,0,...,0,0,0,0,0,0,0,0,0,0


### Últimos pasos de limpieza

Ahora si ya tenemos nuetra base de datos final ***df_final*** solo vamos a hacer 3 pasos adicionales

* Eliminar los registros que se comentaron arriba que incluye (1) personas de edad de 18 o 19 años dado que el filtro de la variable dependiente impactó desproporcionalmente a este grupo (2) Eliminar registros de jovenes de 12 años o más que no tengan data de consumo de drogas ya que se asume que es error de caputa (3) Eliminar el único registro del hombre sin null en la pregunta de embarazo ya que igual se asume que es error de caputa. Esto nos quita un total de solamente 15 registros por lo que nuestra tabla pasa de ser 3040 x 23 a ser 3025 x 23

* Se transforma todo a Int en caso de que tengamos columnas tipo float

* Se crean las sub-bases finales para los 3 modelos que se especifican dados cambios materiales en las variables que existen (1) un modelo para jovenes de 10 y 11 años - sin data de drogas ni embarazo (2) un modelo para jovenes hombres de 12 años a 17 años - sin data de embarazo (3) un modelo para jovenes mujeres de 12 años a 17 años con toda la data

In [1226]:
df_final = df_final[~(df_final['Edad'] >= 18)]
#Should drop 6 rows

df_final = df_final[
    (df_final['Edad'] < 12) |
    (df_final['Opiáceos'].notna())
]
#Should drop 8 rows

df_final = df_final[
    (df_final['Es mujer'] == 1) |
    (df_final['Embarazo'].isna())
]
#Should drop 1 row

df_final = df_final.fillna(0).astype(int)

df_final

/tmp/ipython-input-42175608.py:16: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df_final = df_final.fillna(0).astype(int)


,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente,Robo Agresión o Violencia,Acoso Sexual
0,0,11,1,0,0,0,0,0,0,0,...,0,0,1,0,0,0,0,0,0,0
1,0,15,0,0,1,0,0,0,0,0,...,0,0,0,0,1,1,0,0,0,0
2,0,13,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3,0,10,1,0,0,0,0,0,0,0,...,0,0,0,0,1,0,0,0,0,0
4,0,13,1,0,0,0,0,0,0,0,...,0,0,1,1,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3035,0,13,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3036,0,13,1,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,0
3037,0,11,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3038,0,11,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [1227]:
df_final_10_a_11 = df_final[df_final['Edad'] <= 11]

df_final_10_a_11

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente,Robo Agresión o Violencia,Acoso Sexual
0,0,11,1,0,0,0,0,0,0,0,...,0,0,1,0,0,0,0,0,0,0
3,0,10,1,0,0,0,0,0,0,0,...,0,0,0,0,1,0,0,0,0,0
9,1,11,1,0,0,0,0,0,0,0,...,0,1,1,1,0,0,0,0,0,0
13,0,10,1,0,0,0,0,0,0,0,...,0,1,1,0,0,1,1,0,0,0
15,0,10,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3014,1,10,0,0,0,0,0,0,0,0,...,0,0,1,1,1,0,0,0,0,0
3022,0,11,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3033,0,11,0,0,0,0,0,0,0,0,...,0,0,1,0,0,0,0,0,0,0
3037,0,11,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [1234]:
df_final_12mas_hombre = df_final[(df_final['Edad'] >= 12) & (df_final['Es mujer'] == 0)]

df_final_12mas_hombre

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente,Robo Agresión o Violencia,Acoso Sexual
1,0,15,0,0,1,0,0,0,0,0,...,0,0,0,0,1,1,0,0,0,0
2,0,13,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
5,1,13,0,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,0
6,0,17,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
16,0,15,0,0,0,0,0,0,0,0,...,0,1,0,1,0,0,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3028,0,15,0,0,0,0,0,0,0,0,...,0,0,0,0,1,0,0,0,0,0
3029,0,12,0,0,0,1,0,1,0,0,...,0,0,0,0,0,0,0,0,0,0
3034,0,15,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3035,0,13,0,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [1233]:
df_final_12mas_mujer = df_final[(df_final['Edad'] >= 12) & (df_final['Es mujer'] == 1)]

df_final_12mas_mujer

,Dificultad para aprender,Edad,Es mujer,Fuma,Alcohol,Opiáceos,Tranquilizantes,Anfetaminas,Drogas ilícitas,Diabetes,...,Embarazo,Ansiedad,Depresión,Dificultad de concentración,Dormir mal,Pensamiento suicida,Desordenes alimenticios,Accidente,Robo Agresión o Violencia,Acoso Sexual
4,0,13,1,0,0,0,0,0,0,0,...,0,0,1,1,0,0,0,0,0,0
7,0,17,1,0,0,0,0,0,0,0,...,0,1,0,0,0,0,0,0,0,0
8,0,12,1,0,0,0,0,0,0,0,...,0,1,1,0,0,0,0,0,0,0
10,0,13,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
11,1,15,1,0,0,0,0,0,0,0,...,0,0,0,1,0,1,0,0,0,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3026,0,12,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3030,0,15,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3031,1,14,1,0,0,0,0,0,0,0,...,0,0,1,0,0,0,0,0,0,0
3032,0,14,1,0,0,0,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


## EDA

Con nuestras bases de datos finales para las 3 especificaciones, ahora si podemos a hacer EDA a mayor detalle

ChatGPT fue utilizado como herramienta para corregir código mal escrito y para producir algunas de las lineas de código más complejas